scripts/Tools/pelayout

```diff
@@ -0,0 +1,160 @@
+=======================
+If this tool is missing any feature that you need, please go to github and
+create an issue (https://github.com/ESMCI/cime/issues). Detailed
+information is available at https://github.com/ESMCI/cime.
+
+    COPYRIGHT AND LICENSE
+
+        This library is free software; you can redistribute it and/or modify
+        it under the same terms as Python itself.
+"""
+
+from standard_script_setup import *
+
+from CIME.case import Case
+from CIME.utils import expect, convert_to_string
+import textwrap
+import sys
+import re
+
+logger = logging.getLogger("xmlquery")
+
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+    parser = argparse.ArgumentParser(description="Display and/or change the case's PE layout." ,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     epilog=textwrap.dedent(__doc__) )
+    CIME.utils.setup_standard_logging_options(parser)
+
+    # Set command line options
+    parser.add_argument("-format" , "--format",
+                        help="Format the PE layout items for each component (see below)")
+
+    parser.add_argument("-caseroot" , "--caseroot", default=os.getcwd(),
+                        help="Case directory to build")
+
+    args = parser.parse_args(args[1:])
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    return args.format, args.caseroot
+# End def parse_command_line
+
+
+###############################################################################
+def get_value_as_string(case, var, attribute=None, resolved=False, subgroup=None):
+###############################################################################
+    thistype = case.get_type_info(var)
+    value = case.get_value(var, attribute=attribute, resolved=resolved, subgroup=subgroup)
+    if value is not None and thistype:
+        value = convert_to_string(value, thistype, var)
+    return value
+
+
+reComp = re.compile(r"%%([0-9]*)C")
+reTask = re.compile(r"%%([-+0-9]*)T")
+reThrd = re.compile(r"%%([-+0-9]*)H")
+reRoot = re.compile(r"%%([-+0-9]*)R")
+###############################################################################
+def format_pelayout(comp, ntasks, nthreads, rootpe, format):
+###############################################################################
+    """
+    Format the PE layout information for each component, using a default format,
+    or using the format input, if it exists.
+    """
+    if (format is None):
+        layout_str = "%s: %6d/%d; %6d"%(comp, ntasks, nthreads, rootpe)
+    else:
+        layout_str = re.sub("%", "%%", format)
+        if (reComp.search(layout_str) is not None):
+            layout_str = re.sub(reComp, r"%\1s", layout_str)%comp
+        # End if
+        layout_str = re.sub("%", "%%", layout_str)
```

Comment:
I think there's a typo in your comment. I think the real choices are (for example):

1. Current code: `%C`

2. `%(C)s`

3. `{C}`

I think this is partly a question of UI, and partly a question of code maintainability. I like the idea of supporting user-settable formatting output, and could imagine that idea making its way into other scripts... so the question in my mind is: Do we want to define and parse our own mini-language for this, or just use python's? I'm inclined towards the latter, but I'll be quiet now and let others chime in.